File: from-1701-to-1800/1800-maximum-ascending-subarray-sum/solution.go
func maxAscendingSum(nums []int) int {
    curSum, maxSum := 0, 0
    prev := 0
    for _, curr := range nums {
        if curr > prev {
            curSum += curr
        } else {
            if curSum > maxSum {
                maxSum = curSum
            }
            curSum = curr
        }
        prev = curr
    }
    if curSum > maxSum {
        return curSum
    }
    return maxSum
}

File: from-1701-to-1800/1800-maximum-ascending-subarray-sum/solution.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        maxsum = 0
        cursum = 0
        prev = 0
        for curr in nums:
            if curr > prev:
                cursum += curr
            else:
                if cursum > maxsum:
                    maxsum = cursum
                cursum = curr
            prev = curr
            
        return max(maxsum, cursum)
        